scalyr_agent/builtin_monitors/linux_process_metrics.py

```diff
@@ -619,163 +741,336 @@ class ProcessMonitor(ScalyrMonitor):
     monitor.  However, since ids can change over time, it's better to use the commandline matcher.  The 'pid' field
     is mainly used the linux process monitor run to monitor the agent itself.
 
-    This monitor records the following metrics:
-      app.cpu type=user:                the number of 1/100ths seconds of user cpu time
-      app.cpu type=system:              the number of 1/100ths seconds of system cpu time
-      app.uptime:                       the number of milliseconds of uptime
-      app.threads:                      the number of threads being used by the process
-      app.nice:                         the nice value for the process
-      app.mem.bytes type=vmsize:        the number of bytes of virtual memory in use
-      app.mem.bytes type=resident:      the number of bytes of resident memory in use
-      app.mem.bytes type=peak_vmsize:   the maximum number of bytes used for virtual memory for process
-      app.mem.bytes type=peak_resident: the maximum number of bytes of resident memory ever used by process
-      app.disk.bytes type=read:         the number of bytes read from disk
-      app.disk.requests type=read:      the number of disk requests.
-      app.disk.bytes type=write:        the number of bytes written to disk
-      app.disk.requests type=write:     the number of disk requests.
-      app.io.fds type=open:             the number of file descriptors held open by the process
 
     In additional to the fields listed above, each metric will also have a field 'app' set to the monitor id to specify
     which process the metric belongs to.
 
     You can run multiple instances of this monitor per agent to monitor different processes.
     """
+
+    # poll interval (in seconds) for matching processes when we are monitoring multiple processes
+    # Needs to be initialized as a class variable, not an instance variable so that it is
+    # alive for all epochs of the monitor.
+    LAST_POLLED = {}
```

Comment:
I think it should just be an instance variable.  I'm not quite sure why it would need to be a class var.